packages/catlog/src/dbl/model.rs

```diff
@@ -142,30 +115,32 @@ pub trait DblModel {
 
     /// Acts on a morphism with a morphism operation.
     fn mor_act(&self, m: Self::Mor, α: &Self::MorOp) -> Self::Mor;
+}
 
-    /** Iterates over the basic objects, aka object generators.
-
-    These usually coincide with all of the objects.
-    */
-    fn objects(&self) -> impl Iterator<Item = Self::Ob>;
-
-    /** Iterates over the basic morphisms, aka morphism generators.
-
-    These rarely exhaust all of the morphisms.
-    */
-    fn morphisms(&self) -> impl Iterator<Item = Self::Mor>;
+/// A finitely-generated double model
+pub trait FgDblModel: DblModel + FgCategory {
+    /// get the type of a morphism generator
+    fn mor_gen_type(&self, mor: &Self::MorGen) -> Self::MorType;
 
-    /// Iterates over basic objects of the given type.
-    fn objects_with_type(&self, typ: &Self::ObType) -> impl Iterator<Item = Self::Ob>;
+    /// an iterator over the objects in the model
```

Comment:
Our doc comments should follow the conventions commonly used in the Rust community, including the standard library ([typical examples](https://doc.rust-lang.org/std/vec/struct.Vec.html#implementations)). This seems to include making the first line a short sentence, capitalized and punctuated appropriately, with an active verb.

So this one could be

> Iterates over the objects in the model.

or more verbosely

> Returns an iterator over the objects in the model.

Similarly for the doc comments below.